galley/pkg/config/source/mcp/cache.go

```diff
@@ -174,16 +167,16 @@ func (c *cache) Dispatch(handler event.Handler) {
 	c.handler = event.CombineHandlers(c.handler, handler)
 }
 
-func (c *cache) sync() {
-	if c.synced || !c.started || !c.fullUpdateReceived {
-		return
-	}
-	c.synced = true
+func (c *cache) sync(resources []*resource.Instance) {
```

Comment:
The previous code exists if the cache is already sync'd. The new code performs a full reset sequence for any change to a collection of resources. Is the intent to emulate the resource batching behavior we had before with snapshots?